config/config.go

```diff
@@ -122,16 +125,34 @@ type WarpQuorum struct {
 	QuorumDenominator uint64
 }
 
+// P Chain API configuration, contains the base URL and query parameters
+type PChainAPI struct {
+	BaseURL     string            `mapstructure:"base-url" json:"base-url"`
+	QueryParams map[string]string `mapstructure:"query-parameters" json:"query-parameters"`
+
+	options []rpc.Option
+	client  platformvm.Client
+}
+
+// Info API configuration, contains the base URL and query parameters
+type InfoAPI struct {
+	BaseURL     string            `mapstructure:"base-url" json:"base-url"`
+	QueryParams map[string]string `mapstructure:"query-parameters" json:"query-parameters"`
+
+	options []rpc.Option
+	client  info.Client
+}
```

Comment:
I think it could it be cleaner to have a single `APIConfig` type for the configuration, and keep the actual client instances out of `config` package. It seems a little off to have the `rpc/platformvm/info` packages all be dependencies of `config`.

Reply:
I changed it to `APIConfig` for both P-Chain and Info APIs, then moved the logic of initializing client and options to where it's used in `peers`. Also moved validator client logic to `peers` since they're closely tied.

@najeal Also related to your previous suggestion https://github.com/ava-labs/awm-relayer/pull/268#discussion_r1579897585, implemented a similar change